### ISSUE TITLE
Markdown like-text getting over-formated in table

### ISSUE THREAD

**Issue by JohnMount** [haozhu233/kableExtra#163]
A string of escaped characters that looks like a Markdown link (i.e., things like "&#91;&#93;&#40;&#41;") get rendered as a Markdown link when using kableExtra, but not when knitr alone.

A ".Rmd" file that looks like the following.

````
---
output: html_document
---

```{r}
library("kableExtra")

d <- data.frame(text = "&#91;&#93;&#40;&#41;") # []()
k <- knitr::kable(d, format = "html", escape = FALSE)
kable_styling(k, "striped", full_width = FALSE)
```

````

Produces the output: 

```
<a href></a>
```

And not the expected output:

```
[]()
```

The knitr output alone does produce the expected output.

I know it is a corner case, but some R code (such as array lookups) looks like this when printed.

**Comment by haozhu233** [comment]
Hmmm... Interesting.

On my computer, when I execute the code in R, I can see []() in the preview but after I knit the document, it disappeared and the source code, as you said, turns into `<a href></a>`. I will look into that. Thanks for letting me know. 

My instinct is that it's related to pandoc as it's trying to convert all these things to pure HTML.

**Comment by haozhu233** [comment]
I also found that the direct cause of this problem is that I'm using `xml2` to read in these HTML table, which automatically puts HTML encodings to UTF-8, which in the end got picked up by pandoc. :P 
```
xml2::read_html("<p>&#91;&#93;&#40;&#41;</p>")
# {xml_document}
# <html>
# [1] <body><p>[]()</p></body>
```
I did some research but there doesn't seem to be good solutions as `libxml` is using UTF-8 internally... Hmm, I mean if you don't mind the hacky way, you can do 
```
library("kableExtra")

fix <- function(x) gsub("\\]\\(", "&#93;&#40;", x)

d <- data.frame(text = "&#91;&#93;&#40;&#41;") # []()
k <- knitr::kable(d, format = "html", escape = FALSE)
kable_styling(k, "striped", full_width = FALSE) %>%
  fix()
```
